wgpu-hal/src/metal/adapter.rs

```diff
@@ -276,7 +267,11 @@ impl crate::Adapter<super::Api> for super::Adapter {
             }
         };
 
-        Tfc::COPY_SRC | Tfc::COPY_DST | Tfc::SAMPLED | extra
+        Tfc::COPY_SRC
+            | Tfc::COPY_DST
+            | Tfc::SAMPLED
+            | extra
+            | self.shared.private_caps.sample_count_mask
```

Comment:
This should only be added as part of the `extra` flags as only some formats support MSAA at all